python/tvm/autotvm/task/nnvm_integration.py

```diff
@@ -0,0 +1,291 @@
+# pylint: disable=unused-variable,invalid-name
+"""
+Decorator and utilities for the integration with TOPI and NNVM
+"""
+import warnings
+
+from ... import _api_internal, tensor, placeholder, target as _target
+
+from ..util import get_const_tuple, get_func_name
+from .task import args_to_workload, dispatcher, create, register
+
+# Decorators for registering templates to topi
+# a table that records all registered dispatcher for all targets
+_REGISTED_DISPATHCER = {
+}
+
+def register_topi_compute(topi_compute, target_keys, template_keys):
+    """Register a tunable template for a topi compute function
+
+    Parameters
+    ----------
+    topi_compute: callable
+        The overloaded topi compute call
+    target_keys: str or list of str
+        The compilation target
+    template_keys: str or list of str
+        The template key
+
+    Returns
+    -------
+    decorator: callable
+        A decorator
+    """
+    fname = get_func_name(topi_compute)
+
+    def _decorator(func=None):
```

Comment:
document what is func

Reply:
This default behavior hides quite deep and I am afraid the user may not get it. Try to keep it consistent. For example, always ask the user to pass in func as the last argument of register_topi_compute, if that is None, turn into a decorator.

and not support this default behavior at all